ekncontent/configure.ac

```diff
@@ -0,0 +1,196 @@
+dnl are doing.
+
+# API name and version for pkg-config; increment when breaking API/ABI.
+# This version number is also the major version number.
+m4_define([_EKN_CONTENT_API_VERSION_MACRO], [0])
+
+# Minor and micro versions: increment micro version when making a release. Minor
+# version is even for a stable release and odd for a development release.
+# When making any release, if the API changes, set the interface age to 0.
+m4_define([_EKN_CONTENT_MINOR_VERSION_MACRO], [0])
+m4_define([_EKN_CONTENT_MICRO_VERSION_MACRO], [0])
+m4_define([_EKN_CONTENT_INTERFACE_AGE_MACRO], [0])
+
+# Full version, for use in AC_INIT
+m4_define([_EKN_CONTENT_VERSION_MACRO],
+    [_EKN_CONTENT_API_VERSION_MACRO._EKN_CONTENT_MINOR_VERSION_MACRO._EKN_CONTENT_MICRO_VERSION_MACRO])
+
+
+# Initialization
+# --------------
+# Initialize Autoconf: package name, version, bug report address, tarball name,
+# website
+AC_INIT([Endless OS Knowledge Content Library], [_EKN_CONTENT_VERSION_MACRO],
+    [], [eos-knowledge-content], [http://endlessm.com])
+# Verify that the source directory can be found
+AC_CONFIG_SRCDIR([ekncontent/eknc-utils.h])
+# Initialize Automake: enable all warnings and do not insist on GNU standards
+# no-portability suppresses warnings about syntax specific to GNU make
+# parallel-tests specifies that we use the new parallel-running test harness.
+# Unlike serial-tests, this option is accepted by Automake 1.11
+# tar-ustar is required because there may be files whose entire paths exceed
+# 99 characters.
+# subdir-objects is for forward compatibility with Automake 2.0 and can be
+# removed when we upgrade to that version.
+AM_INIT_AUTOMAKE([-Wall -Wno-portability foreign 1.11 parallel-tests tar-ustar subdir-objects])
+# Avoid spewing garbage over the terminal ('make V=1' to see the garbage)
+AM_SILENT_RULES([yes])
+# Initialize Libtool; don't build static libraries
+LT_INIT([disable-static])
+# Keep Autotools macros local to this source tree
+AC_CONFIG_MACRO_DIR([m4])
+
+AC_CACHE_SAVE
+
+# Variables to define
+# -------------------
+# Version numbers and macros
+EKN_CONTENT_API_VERSION=_EKN_CONTENT_API_VERSION_MACRO
+AC_SUBST(EKN_CONTENT_API_VERSION)
+EKN_CONTENT_API_NAME="eos-knowledge-content-$EKN_CONTENT_API_VERSION"
+AC_SUBST(EKN_CONTENT_API_NAME)
+EKN_CONTENT_INTERFACE_AGE=_EKN_CONTENT_INTERFACE_AGE_MACRO
+AC_SUBST([EKN_CONTENT_MAJOR_VERSION], [_EKN_CONTENT_API_VERSION_MACRO])
+AC_SUBST([EKN_CONTENT_MINOR_VERSION], [_EKN_CONTENT_MINOR_VERSION_MACRO])
+AC_SUBST([EKN_CONTENT_MICRO_VERSION], [_EKN_CONTENT_MICRO_VERSION_MACRO])
+AC_DEFINE([EKN_CONTENT_VERSION], [_EKN_CONTENT_VERSION_MACRO], [Full version number])
+
+# Shared library versioning; calculated automatically.
+# See http://www.gnu.org/software/libtool/manual/html_node/Updating-version-info.html
+EKN_CONTENT_LT_CURRENT=m4_eval(100 * _EKN_CONTENT_MINOR_VERSION_MACRO + _EKN_CONTENT_MICRO_VERSION_MACRO - _EKN_CONTENT_INTERFACE_AGE_MACRO)
```

Comment:
I'd go with https://developer.gnome.org/programming-guidelines/stable/versioning.html.en#libtool-versioning instead of this calculation